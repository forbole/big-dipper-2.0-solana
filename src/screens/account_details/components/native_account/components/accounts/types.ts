export type AddressesType = string[];

export type AccountsType = {
  stake: AddressesType;
  nonce: AddressesType;
  vote: AddressesType;
  token: AddressesType;
}
