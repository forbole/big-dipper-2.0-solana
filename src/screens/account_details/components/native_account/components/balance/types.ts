export type BalanceType = {
  native: TokenUnit;
  stake: TokenUnit;
  nonce: TokenUnit;
  vote: TokenUnit;
  total: TokenUnit;
}
