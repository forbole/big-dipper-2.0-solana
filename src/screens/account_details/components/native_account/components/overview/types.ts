export type OverviewType = {
  address: string;
}
