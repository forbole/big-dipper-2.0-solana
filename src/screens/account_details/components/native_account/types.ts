export type OverviewType = {
  address: string;
}

export type AccountType = {
  address: string;
}

export type AccountsType = {
  data: {
    [value:number]: AccountType[];
  };
  count: number;
  loading: boolean;
}

export type NativeAccountState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
}
