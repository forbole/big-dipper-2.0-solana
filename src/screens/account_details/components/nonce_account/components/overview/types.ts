export type OverviewType = {
  address: string;
  balance: number;
  authority: string;
  blockhash: string;
  fee: number;
}
