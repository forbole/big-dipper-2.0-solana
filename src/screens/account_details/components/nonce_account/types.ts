export type OverviewType = {
  address: string;
  authority: string;
  balance: TokenUnit;
  blockhash: string;
  fee: number;
}

export type NonceAccountState = {
  loading: boolean;
  overview: OverviewType;
}
