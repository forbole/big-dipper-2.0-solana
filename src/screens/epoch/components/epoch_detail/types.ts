export type EpochDetailType = {
    name: string;
    value: number | string;
    detail: string;
  }
