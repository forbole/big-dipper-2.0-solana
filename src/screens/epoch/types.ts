export type InflationType = {
  validator: number;
  foundation: number;
  total: number;
  epoch: number;
}

export type InflationGovernorType = {
  initial: number;
  terminal: number;
  taper: number;
  foundation: number;
  foundationTerm: number;
}

export type EpochScheduleType = {
  slotsPerEpoch: number;
  leaderScheduleSlotOffset: number;
  warmup: boolean;
  firstNormalEpoch: number;
  firstNormalSlot: number;
}
