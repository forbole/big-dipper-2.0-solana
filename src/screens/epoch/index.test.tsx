/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import {
  MockTheme, wait,
} from '@tests/utils';
import { EpochDetailsDocument } from '@graphql/types';
import Epoch from '.';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@components', () => ({
  Layout: (props) => <div id="Layout" {...props} />,
  LoadAndExist: (props) => <div id="LoadAndExist" {...props} />,
}));

jest.mock('./components', () => ({
  EpochDetail: (props) => <div id="EpochDetail" {...props} />,
}));

const mockEpochDetailsDocument = jest.fn().mockResolvedValue({
  "data": {
    "inflationRate": {
      "epoch": 305,
      "foundation": 0,
      "total":0.06963020582432754,
      "validator": 0.06963020582432754,
    },
    "inflationGovernor": {
      "foundation": 0,
      "foundationTerm": 0,
      "initial": 0.08,
      "taper": 0.15,
      "terminal": 0.015,
    },
    "epochSchedule": {
      "firstNormalEpoch": 0,
      "firstNormalSlot": 0,
      "leaderScheduleSlotOffset": 432000,
      "slotsPerEpoch": 432000,
      "warmup": false,
    }
  }
});


// ==================================
// unit tests
// ==================================
describe('screen: Epoch', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();

    mockClient.setRequestHandler(
      EpochDetailsDocument,
      mockEpochDetailsDocument,
    );

console.log('mockEpochDocument', mockEpochDetailsDocument)

    let component;

    renderer.act(() => {
      component = renderer.create(
          <ApolloProvider client={mockClient}>
            <MockTheme>
              <Epoch />
            </MockTheme>
          </ApolloProvider>
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
