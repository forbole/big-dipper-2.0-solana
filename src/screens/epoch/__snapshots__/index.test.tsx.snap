// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Epoch matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Layout"
  navTitle="epoch:epoch"
>
  <div
    exists={true}
    id="LoadAndExist"
    loading={false}
  >
    <div
      epochDetail={
        Array [
          Object {
            "detail": "epoch:epochDetail",
            "name": "epoch:epoch",
            "value": "305",
          },
          Object {
            "detail": "epoch:totalInflationDetail",
            "name": "epoch:totalInflation",
            "value": "6.96%",
          },
          Object {
            "detail": "epoch:validatorInflationDetail",
            "name": "epoch:validatorInflation",
            "value": "6.96%",
          },
          Object {
            "detail": "epoch:foundationInflationDetail",
            "name": "epoch:foundationInflation",
            "value": "0%",
          },
        ]
      }
      id="EpochDetail"
      title="inflation"
    />
    <div
      epochDetail={
        Array [
          Object {
            "detail": "epoch:initialDetail",
            "name": "epoch:initial",
            "value": "8%",
          },
          Object {
            "detail": "epoch:terminalDetail",
            "name": "epoch:terminal",
            "value": "1.5%",
          },
          Object {
            "detail": "epoch:taperDetail",
            "name": "epoch:taper",
            "value": "15%",
          },
          Object {
            "detail": "epoch:foundationDetail",
            "name": "epoch:foundation",
            "value": "0%",
          },
          Object {
            "detail": "epoch:foundationTermDetail",
            "name": "epoch:foundationTerm",
            "value": "0",
          },
        ]
      }
      id="EpochDetail"
      title="inflationGovernor"
    />
    <div
      epochDetail={
        Array [
          Object {
            "detail": "epoch:slotsPerEpochDetail",
            "name": "epoch:slotsPerEpoch",
            "value": "432,000",
          },
          Object {
            "detail": "epoch:leaderScheduleSlotOffsetDetail",
            "name": "epoch:leaderScheduleSlotOffset",
            "value": "432,000",
          },
          Object {
            "detail": "epoch:warmupDetail",
            "name": "epoch:warmup",
            "value": "FALSE",
          },
          Object {
            "detail": "epoch:firstNormalEpochDetail",
            "name": "epoch:firstNormalEpoch",
            "value": "0",
          },
          Object {
            "detail": "epoch:firstNormalSlotDetail",
            "name": "epoch:firstNormalSlot",
            "value": "0",
          },
        ]
      }
      id="EpochDetail"
      title="epochSchedule"
    />
  </div>
</div>
`;
